skimage/_shared/tests/test_utils.py

```diff
@@ -152,24 +144,24 @@ def bar(arg0, new_arg1=1, arg2=None):
 
     # Assert that nothing happens when the function is called with the
     # new API
-    with pytest.warns(None) as record:
```

Comment:
Don't we still need this pytest warning assertion?

Reply:
Well, it should fail for the CI job with strict warnings enabled. To keep it equivalent to before, let me change it to use `warnings.catch_warnings` so we can still demonstrate that there are none raised

Reply:
see a49e9b5 for updated code that still checks the recorded warnings to make sure there are no